mlrun/api/db/sqldb/db.py

```diff
@@ -3030,3 +3030,142 @@ def create_data_version(self, session, version):
         now = datetime.now(timezone.utc)
         data_version_record = DataVersion(version=version, created=now)
         self._upsert(session, [data_version_record])
+
+    @retry_on_conflict
+    def store_background_task(
+        self,
+        session,
+        name: str,
+        state: str = mlrun.api.schemas.BackgroundTaskState.running,
+        project: str = None,
+        timeout: int = None,
+    ):
+        background_task_record = (
+            self._query(
+                session,
+                BackgroundTask,
+                name=name,
+                project=project,
+            )
+            .filter(BackgroundTask.project == project)
+            .one_or_none()
+        )
+        now = datetime.now(timezone.utc)
+        if background_task_record:
+            # we don't want to be able to change state after it reached terminal state
+            if (
+                background_task_record.state
+                in mlrun.api.schemas.BackgroundTaskState.terminal_states()
+                and state != background_task_record.state
+            ):
+                raise mlrun.errors.MLRunRuntimeError(
+                    "Background task already exists with terminal state. failing"
+                )
+            # we don't set default background task timeout, to be able to resolve the timeout when the API
+            # queries the record, so even if the API restarts with new default timeout it will apply the new timeout
+            background_task_record.timeout = (
+                timeout
+                if timeout and not background_task_record.timeout
+                else background_task_record.timeout
+            )
+            background_task_record.state = state
+            background_task_record.updated = now
+        else:
+            background_task_record = BackgroundTask(
+                name=name,
+                project=project,
+                state=mlrun.api.schemas.BackgroundTaskState.running,
+                created=now,
+                updated=now,
+                timeout=timeout,
+            )
+        self._upsert(session, [background_task_record])
+
+    def get_background_task(
+        self, session, name: str, project: str = None
+    ) -> schemas.BackgroundTask:
+        background_task_record = self._get_background_task_record(
+            session, name, project
+        )
+
+        return self._transform_background_task_record_to_schema(background_task_record)
+
+    @staticmethod
+    def _transform_background_task_record_to_schema(
+        background_task_record: BackgroundTask,
+    ) -> schemas.BackgroundTask:
+        timeout = background_task_record.timeout
+        if mlrun.mlconf.background_tasks_timeout_defaults.mode == "enabled":
+            timeout = (
+                timeout
+                if timeout
+                else mlrun.mlconf.background_tasks_timeout_defaults.default
+            )
```

Comment:
I don't think this is needed, timeout is being set on save, no need to add it on get